### ISSUE TITLE
Memory leak when using akka instrumentation

### ISSUE THREAD

**Issue by gipeshka** [kamon-io/Kamon#886]
Hi folks!

While using Kamon instrumentation with our services we have spotted a memory leak on long-living instances. The service itself is a simple message-bus consumer built on top of play framework. The service interacts with multiple other services using akka-http.

The object dominating the heap is an `akka.actor.ActorCell` of `akka.stream.impl.StreamSupervisor`. The supervisor contains an `akka.actor.dungeon.ChildrenContainer$NormalChildrenContainer` with a huge treemap of `akka.actor.ChildRestartStats` of sinks (according to the name) like `flow-123456789-0-ignoreSink`. The actors that are reporting stats are `application/system/MaterializerGuardian/StreamSupervisor/ActorGraphInterpreter` lying under `StreamSupervisor-0`.

The only manually created ignore sinks that I was able to find in our app were akka http entity's `.discardBytes`. Though there definitely can be many more implicit ignore-sinks in PlayFramework or akka http themselves. 

To verify the case we've turned off actor instrumentation for some time and now there's no memory leak and the `akka.actor.ActorCell` of `akka.stream.impl.StreamSupervisor` is not present according to fresh heap dumps.

Versions:
Kamon version: 2.1.6. We only use `kamon-akka-http`.
Play framework version: 2.8.2
Akka version: 2.6.9
Akka http version: 4.0.1

**Comment by SimunKaracic** [comment]
What kind of filters are you using for actors? Which ones are you including and excluding?

**Comment by gipeshka** [comment]
The default ones, no customization.

**Comment by ivantopo** [comment]
Hey @gipeshka! Can you help us create a reproducible for this issue? and also, would it be possible to take share that heap dump privately? (you can email me via ivantopo@kamon.io to coordinate that). Thanks!

**Comment by gipeshka** [comment]
Hi @ivantopo, I shared the dump with you. Reproducing the issue is not that simple, as the leak seemed to be quite slow (it took our service about a week to fill the heap completely). Currently we've disabled akka.actor instrumentation in prod, other envs unfortunately don't have enough load. I also wasn't able to reproduce that locally.

**Comment by tupol** [comment]
I have seen this behavior, and I confirm that removing the Akka HTTP instrumentation solved the memory leak. 

We bumped into this by running Jmeter to evaluate the performance impact of Kamon on an application :) (days of runtime in a few minutes)

**Comment by calvernaz** [comment]
@tupol @gipeshka could you show how did you disabled the Akka HTTP instrumentation?

**Comment by gipeshka** [comment]
```
kanela.modules {
  akka.enabled = no
  akka-remote.enabled = no
}
```

**Comment by ivantopo** [comment]
Hey @gipeshka and folks! 

Not sure if you noticed it, but we just released Kamon `2.1.11` with a fix to this issue. It took quite a lot of digging and at the end, the issue had nothing to do with Akka HTTP itself, but rather with a bug on the Actor instrumentation that could prevent some system messages from being processed, which in some cases would lead to actors never being stopped.

Please let us know if you ever see this issue again!
